src/main/java/seedu/address/logic/commands/EditStudentCommand.java

```diff
@@ -235,7 +247,8 @@ public boolean equals(Object other) {
                     && getAddress().equals(e.getAddress())
                     && getStudyLevel().equals(e.getStudyLevel())
                     && getGuardianPhone().equals(e.getGuardianPhone())
-                    && getRelationship().equals(e.getRelationship());
+                    && getRelationship().equals(e.getRelationship())
+                    && getSessions().equals(e.getSessions());
```

Comment:
Might not need to check for same sessions?

Reply:
Amended in next commit!